### PR TITLE
Document AutoChainRules

### PR DESCRIPTION
**Pull request by gdalle**
- Add docstring to `AutoChainRules`
- Add `AutoChainRules` to README

### CODE REVIEW

**Comment by gdalle** [comment]
Once this is merged, would it be possible to tag a new release?

File: src/ADTypes.jl [hunk 1]
```diff
@@ -19,6 +19,15 @@ abstract type AbstractSparseReverseMode <: AbstractReverseMode end
...
+    AutoChainRules{RC}
+
+Chooses any AD library based on [ChainRulesCore.jl](https://github.com/JuliaDiff/ChainRulesCore.jl), given an appropriate [`RuleConfig`](https://juliadiff.org/ChainRulesCore.jl/stable/rule_author/superpowers/ruleconfig.html) object.
+
+# Fields
+
+- `ruleconfig::RC`
+"""
 Base.@kwdef struct AutoChainRules{RC} <: AbstractADType
     ruleconfig::RC
```
Comment (Vaibhavdixit02) [inline src/ADTypes.jl:32]: Can this have a default value?

**Comment by Vaibhavdixit02** [comment]
Feel free to bump the version in this PR itself

File: src/ADTypes.jl (hunk 1 above)
Comment (gdalle) [reply src/ADTypes.jl:32]: Not without loading an extension

**Comment by gdalle** [comment]
> Feel free to bump the version in this PR itself

Done

**Comment by gdalle** [comment]
But we should probably wait until I address #25 before releasing 0.2.7, that way we will have a nice documentation too

File: src/ADTypes.jl (hunk 1 above)
Comment (Vaibhavdixit02) [reply src/ADTypes.jl:32]: Okay, makes sense

**Pull request merged**